sdk/ga4/transform/index.ts

```diff
@@ -0,0 +1,106 @@
+import type {
+  ToAnalyticsProps,
+  ToAddToCart,
+  ToRemoveFromCart,
+  ToViewItem,
+  ToSelectPromotion,
+  AnalyticsEventTypes,
+} from "../types/index.ts"
+
+import { toAnalyticsItems, getTotalValue } from "./toAnalyticsItems.ts"
+
+type HandlerReturnTypes = {
+  add_to_cart: ReturnType<typeof toAddToCart>
+  remove_from_cart: ReturnType<typeof toRemoveFromCart>
+  view_item: ReturnType<typeof toViewItem>
+  select_promotion: ReturnType<typeof toSelectPromotion>
+}
+
+const toAddToCart = (data: ToAddToCart) => {
+  const items = toAnalyticsItems({ items: data.items, extended: data.extended })
+  const value = getTotalValue(items)
+
+  return {
+    name: "add_to_cart" as const,
+    params: {
+      currency: data.extended.currency ?? "BRL",

```

Comment:
if this is required, why is it inside the extension part?

Reply:
The idea would be to keep items only with data that, within the Deco product, currency is not present, so it would be sent separately in cases of items, for example, orderform vtex